custom_build.sh

```diff
@@ -0,0 +1,191 @@
+    marlin_url_2="https://github.com/bourgesl/marlin-renderer/releases/download/v${marlin_major}_${marlin_minor}_${marlin_patch}/marlin-${marlin_major}.${marlin_minor}.${marlin_patch}-Unsafe-sun-java2d.jar"
+    download_from_url_to_a_filepath "${marlin_url_1}" "${MARLIN_ARTIFACT_DIRECTORY}/marlin-${marlin_major}.${marlin_minor}.${marlin_patch}-Unsafe.jar"
+    download_from_url_to_a_filepath "${marlin_url_2}" "${MARLIN_ARTIFACT_DIRECTORY}/marlin-${marlin_major}.${marlin_minor}.${marlin_patch}-Unsafe-sun-java2d.jar"
+}
+
+function download_geoserver() {
+    clean_up_directory ${GEOSERVER_ARTIFACT_DIRECTORY}
+    local VERSION=${1}
+    local GEOSERVER_FILE_NAME_NIGHTLY="geoserver-${VERSION}-latest-war.zip"
+		local GEOSERVER_FILE_NAME_STABLE="geoserver-${VERSION}-war.zip"
+
+		if [[ ( "${VERSION}" =~ "x" ) || ( "${VERSION}" == "master" ) ]]; then
+			local GEOSERVER_ARTIFACT_URL=${BASE_BUILD_URL}/${VERSION}/${GEOSERVER_FILE_NAME_NIGHTLY}
+		else
+			local GEOSERVER_ARTIFACT_URL=${BASE_BUILD_URL_STABLE}/${VERSION}/${GEOSERVER_FILE_NAME_STABLE}
+		fi
+
+    if [ -f /tmp/geoserver.war.zip ]; then
+        rm /tmp/geoserver.war.zip
+    fi
+    if [ ! -e "${GEOSERVER_ARTIFACT_DIRECTORY}" ]; then
+        mkdir -p "${GEOSERVER_ARTIFACT_DIRECTORY}"
+    fi
+    download_from_url_to_a_filepath  "${GEOSERVER_ARTIFACT_URL}" "/tmp/geoserver.war.zip"
+    unzip -p /tmp/geoserver.war.zip geoserver.war > ${GEOSERVER_ARTIFACT_DIRECTORY}/geoserver.war
+}
+
+
+function build_with_data_dir() {
+
+	local TAG=${1}
+        local PULL_ENABLED=${2}
+        if [[ "${PULL_ENABLED}" == "pull" ]]; then
+                DOCKER_BUILD_COMMAND="docker build --pull"
+        else
+                DOCKER_BUILD_COMMAND="docker build"
+        fi;
+	${DOCKER_BUILD_COMMAND} --no-cache \
+		--build-arg GEOSERVER_WEBAPP_SRC=${GEOSERVER_ARTIFACT_DIRECTORY}/geoserver.war \
+    --build-arg PLUG_IN_URLS=$PLUGIN_ARTIFACT_DIRECTORY \
+    --build-arg GEOSERVER_DATA_DIR_SRC=${GEOSERVER_DATA_DIR_DIRECTORY} \
+		-t geosolutionsit/geoserver:"${TAG}-${GEOSERVER_VERSION}" \
+		 .
+}
+
+function build_without_data_dir() {
+
+	local TAG=${1}
+	local PULL_ENABLED=${2}
+	if [[ "${PULL_ENABLED}" == "pull" ]]; then
+		DOCKER_BUILD_COMMAND="docker build --pull"
+	else
+		DOCKER_BUILD_COMMAND="docker build"
+	fi;
+	${DOCKER_BUILD_COMMAND} --no-cache \
+		--build-arg GEOSERVER_WEBAPP_SRC=${GEOSERVER_ARTIFACT_DIRECTORY}/geoserver.war \
+    --build-arg PLUG_IN_URLS=$PLUGIN_ARTIFACT_DIRECTORY\
+		-t geosolutionsit/geoserver:"${TAG}-${GEOSERVER_VERSION}" \
+		 .
+}
+
+function main {
+    help ${ALL_PARAMETERS}
+    download_geoserver "${GEOSERVER_VERSION}"
+    clean_up_directory ${PLUGIN_ARTIFACT_DIRECTORY}
+    download_plugin ext monitor
+    download_plugin ext control-flow
+    download_plugin community sec-oauth2-geonode
```

Comment:
i had to comment out all plugin downloads to get this running on my machine

Reply:
they were there just as examples but they should work, I'd need output of:
```
bash -x ./custom_build.sh <options>
```
to understand well